HelpSource/Classes/Delay1.schelp

```diff
@@ -24,17 +24,27 @@ This value will be added to the output.
 
 argument::x1
 The sample value preceding the delayed input, which is also the value of the first output sample.
-This is a numeric value, a code::UGen::, or code::nil::.
-A value of code::nil:: is equivalent to setting strong::x1:: to strong::in::, effectively repeating the first input sample once.
+
+This can be a numeric value or a code::UGen::.
+code::nil:: is also supported, which is equivalent to setting code::x1: in::, effectively repeating the first sample of the input signal.
```

Comment:
If you like, you could write more simply:

If code::x1:: is unspecified (code::nil::), it is internally set to code::x1 = in::, effectively repeating the first sample of the input signal.

Reply:
Yes, this is better wording.

However, with your suggested change to remove `?? { in }`, then `nil` would no longer be supported at audio rate (the UGen would fail to compile if `nil` were passed in), so this description wouldn't be accurate. 

So there would have to be two separate descriptions, one for audio rate and one for control rate. This was the reason for suggesting that `?? { in }` remain in, since it has no practical impact users (it actually gives them more options, even if it's seldom used), and, most importantly IMO, simplifies communication about the behavior.

I don't outright object to removing `?? { in }`, I just want to make sure my intention was clear before trying to rework the docs again.

Reply:
I see your point, it is more uniform from the perspective of argument values. I would look at it from the perspective of default values. Currently, in the kr case, the default is `in`, and in the ar case, the default is `0`, _unless_ you actively replace it with `nil`. I think this makes things hard to understand. It would be clearer if this "unless" could be dropped.

Btw it could be even made clearer if you'd write
```
*kr { arg in = 0.0, mul = 1.0, add = 0.0, x1 = (in);
```

Reply:
> Btw it could be even made clearer if you'd write
*kr { arg in = 0.0, mul = 1.0, add = 0.0, x1 = (in);

Ah, this is great! I didn't know it was possible to use argument names when specifying programatic defaults.

This actually solves the issue completely, as `nil` was only being used to handle this default case. So now we can just remove support for `nil` as an argument (unless there's some other reason to keep it?). I'll update the class, docs, and tests in the next day or two.

Reply:
yes, so it would read clearly:
```
*ar { arg in = 0.0, mul = 1.0, add = 0.0, x1 = 0.0, x2 = 0.0;
*kr { arg in = 0.0, mul = 1.0, add = 0.0, x1 = (in), x2 = (in);
```

and all the `x1 = x1 ?? { in }` can go away.

Reply:
I've now removed all the logic related to supporting `nil`, updated the docs, and the unit tests.